calour/heatmap/heatmap.py

```diff
@@ -238,49 +283,21 @@ def heatmap(exp: Experiment, sample_field=None, feature_field=None,
         ax.xaxis.set_visible(False)
     else:
         try:
-            xticks = _transition_index(exp.sample_metadata[sample_field])
+            ticklabels = exp.sample_metadata[sample_field]
         except KeyError:
             raise ValueError('Sample field %r not in sample metadata' % sample_field)
-        tick_pos, tick_val = zip(*xticks)
-        tick_pos = np.array([0.] + list(tick_pos))
-        # samples position - 0.5 before and go to 0.5 after
-        tick_pos -= 0.5
-        for pos in tick_pos[1:-1]:
-            ax.axvline(x=pos, color='white', linewidth=1)
-        # set tick/label at the middle of each sample group
-        ax.set_xticks(tick_pos[:-1] + (tick_pos[1:] - tick_pos[:-1]) / 2)
-        if xticklabel_kwargs is None:
-            xticklabel_kwargs = {'rotation': 45, 'ha': 'right'}
-        ax.set_xticklabels(_truncate_middle(tick_val, xticklabel_len), **xticklabel_kwargs)
+        _set_axis_ticks(ax, 'x', ticklabels, xticks_max, numcols, xticklabel_kwargs, xticklabel_len)
         ax.set_xlabel(sample_field)
 
     # plot y tick labels dynamically
     if feature_field is None:
         ax.yaxis.set_visible(False)
     else:
         try:
-            yticklabels = _truncate_middle(exp.feature_metadata[feature_field], yticklabel_len)
+            ticklabels = exp.feature_metadata[feature_field]
         except KeyError:
             raise ValueError('Feature field %r not in feature metadata' % feature_field)
-        if max_yticks is None:
-            # show all labels
-            max_yticks = numcols
-        # should be not larger than maxticks
-        max_yticks = min(max_yticks, mpl.ticker.Locator.MAXTICKS)
-
-        def format_fn(tick_val, tick_pos):
-            # cf http://matplotlib.org/gallery/ticks_and_spines/tick_labels_from_values.html
-            if 0 <= tick_val < numcols:
-                return yticklabels[int(tick_val)]
-            else:
-                return ''
-        # set the maximal number of feature labels
-        ax.yaxis.set_major_formatter(mpl.ticker.FuncFormatter(format_fn))
-        ax.yaxis.set_major_locator(mpl.ticker.MaxNLocator(max_yticks, integer=True))
-        if yticklabel_kwargs is None:
-            yticklabel_kwargs = {'rotation': 45, 'ha': 'right', 'size': 6}
-        for t in ax.get_yticklabels():
-            t.set(**yticklabel_kwargs)
+        _set_axis_ticks(ax, 'y', ticklabels, yticks_max, numrows, yticklabel_kwargs, xticklabel_len)
```

Comment:
isn't it supposed to be yticklabel_len and not xticklabel_len?

Reply:
typo. fixed